### PR TITLE
APPEALS-8386 Calculation Functionality for Timely/Untimely Checkbox on DAS

### PR DESCRIPTION
**Pull request by zurbergram**
Resolves https://vajira.max.gov/browse/APPEALS-8386

### Description
On Evaluate Decision page update Case Timeliness section to include an Attorney Case Timeline and update calculations

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Post this PR in #appeals-schema with a summary
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

File: app/views/queue/index.html.erb [hunk 1]
```diff
@@ -46,7 +46,8 @@
       vha_irregular_appeals: FeatureToggle.enabled?(:vha_irregular_appeals, user: current_user),
       vso_virtual_opt_in: FeatureToggle.enabled?(:vso_virtual_opt_in, user: current_user),
       das_case_timeliness: FeatureToggle.enabled?(:das_case_timeliness, user: current_user),
-      split_appeal_workflow: FeatureToggle.enabled?(:split_appeal_workflow, user: current_user)
+      split_appeal_workflow: FeatureToggle.enabled?(:split_appeal_workflow, user: current_user),
+      das_case_timeline: FeatureToggle.enabled?(:das_case_timeline, user: current_user),
```
Comment (harishvanga333) [inline app/views/queue/index.html.erb:50]: end tag, can you remove ","

**Review by jshields13 — APPROVED** [review]

File: app/views/queue/index.html.erb (hunk 1 above)
Comment (zurbergram) [reply app/views/queue/index.html.erb]: done

**Review by harishvanga333 — APPROVED** [review]

File: client/app/queue/selectors.js [hunk 2]
```diff
@@ -229,6 +231,68 @@ export const camoAssignTasksSelector = createSelector(
...
+    const attorneyTasks = filter(tasks, (task) => task.parentId == parentId && types.includes(task.type));
+
+    // eslint-disable-next-line id-length
+    attorneyTasks.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt));
+
+    return attorneyTasks;
+  }
+);
+
+// Get all task trees for all Attorney Type Tasks found with the JudgeDecisionReviewTaskId as their parentId
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply client/app/queue/selectors.js:251]: I was losing track of how it all worked, lol

**Review by kristeja — APPROVED** [review]
🚀

**Pull request merged**